nixos/tests/matrix/matrix-alertmanager.nix

```diff
@@ -0,0 +1,190 @@
+
+    nodes = {
+      homeserver =
+        { pkgs, ... }:
+        {
+          services.matrix-synapse = {
+            enable = true;
+            settings = {
+              database.name = "sqlite3";
+              tls_certificate_path = "${cert}";
+              tls_private_key_path = "${key}";
+              enable_registration = true;
+              enable_registration_without_verification = true;
+              registration_shared_secret = "supersecret-registration";
+
+              listeners = [
+                {
+                  # The default but tls=false
+                  bind_addresses = [
+                    "0.0.0.0"
+                  ];
+                  port = 8448;
+                  resources = [
+                    {
+                      compress = true;
+                      names = [ "client" ];
+                    }
+                    {
+                      compress = false;
+                      names = [ "federation" ];
+                    }
+                  ];
+                  tls = false;
+                  type = "http";
+                  x_forwarded = false;
+                }
+              ];
+            };
+          };
+
+          networking.firewall.allowedTCPPorts = [ 8448 ];
+
+          environment.systemPackages = [
+            (pkgs.writeShellScriptBin "register_alertmanager_user" ''
+              exec ${pkgs.matrix-synapse}/bin/register_new_matrix_user \
+                -u alertmanager \
+                -p alertmanager-password \
+                --admin \
+                --shared-secret supersecret-registration \
+                http://localhost:8448
+            '')
+            # This is needed to solve a chicken and egg
+            # problem. Matrix-alertmanager expects a token for authentication,
+            # but a token is create after the user has been registered. This
+            # changes the token in the database to match the one specified in
+            # the service settings.
+            (pkgs.writers.writePython3Bin "hardcode_matrix_values"
+              {
+                libraries = with pkgs.python3Packages; [
+                  sqlite-utils
+                ];
+              }
+              ''
+                import sqlite3
+                con = sqlite3.connect("/var/lib/matrix-synapse/homeserver.db")
+                cur = con.cursor()
+                cur.execute("update access_tokens set token='faketoken' where id = 2")
+                con.commit()
+                con.close()
+              ''
+            )
+          ];
+        };
+
+      matrix_alertmanager =
+        { pkgs, ... }:
+        {
+          services.matrix-alertmanager = {
+            enable = true;
+            tokenFile = pkgs.writeText "token.txt" "faketoken";
+            secretFile = pkgs.writeText "secret.txt" "wontbeused";
+            homeserverUrl = "http://homeserver:8448";
+            matrixRooms = "matrix/!123:localhost:8448";
```

Comment:
I'm not very familiar with matrix. Is "123" a room name? How does this relate to the "alertmanager" room you create below in "create_room_and_join"?

Reply:
Added a comment for this. This is a [Room ID in matrix terminology since it starts with a `!`](https://spec.matrix.org/v1.13/#room-structure). Matrix-alertmanager accepts room ids and not room aliases.

I'll change this to make it more explicit that it's a room id (the module documentation covers this) in the test as well. But it's not used anywhere currently, it's only needed for the bot to start.